avail-server/src/main/java/com/avail/server/AvailServer.kt

```diff
@@ -595,9 +597,15 @@ class AvailServer constructor(
 		recordUpgradeRequest(channel,uuid) {
 				upgradedChannel, receivedUUID, resumeUpgrader ->
 			assert(uuid == receivedUUID)
+			val oldId = upgradedChannel.id
+			upgradedChannel.id = receivedUUID
+			upgradedChannel.parentId = channel.id
 			upgradedChannel.upgradeToIOChannel()
 			resumeUpgrader()
 			afterUpgraded(upgradedChannel)
+			logger.log(
+				Level.FINEST,
+				"Channel, $oldId upgraded to: $upgradedChannel")
```

Comment:
This would probably be better as:

```kt
"Channel [$oldId] upgraded to: [$upgradedChannel]"
```

To better visually encapsulate the UUIDs, but I won't hold up a merge for this.